core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableHandle.java

```diff
@@ -13,10 +13,29 @@
  */
 package io.trino.spi.connector;
 
+import java.util.Random;
+
 /**
  * Represents a handle to a relation returned from the connector to the engine.
  * It will be used by the engine whenever given relation will be accessed.
  */
 public interface ConnectorTableHandle
 {
+    /**
+     * Returns true if a ConnectorTableHandle is "fusible" with another ConnectorTableHandler.
+     * Fusible here means the two tables can possibly be optimized by SharedScanOptimizer.
+     */
+    default boolean equalsForFusion(Object o)
```

Comment:
It seems better to add this to `ConnectorMetadata`, similar to the existing `getCommonPartitioningHandle()` method. Since this isn't a generic equals method, the argument can be `ConnectorTableHandle` rather than `Object`. The engine would guarantee that it only calls this method with handle instances for the same connector.

Reply:
Thanks for your review! Currently in this optimization rule the focus mainly surrounds the `TableScanNode` plan nodes and scrutinizes their eligibility for being fused. `TableScanNode` consists of a `TableHandle`, which consists of a `ConnectorTableHandle` that seems to be natural for the abstraction. Can you please shed more light on how this will work with the `ConnectorMetadata` approach?